test/unit/transaction-runner-test.coffee

```diff
@@ -327,20 +328,63 @@ describe 'TransactionRunner', ()->
           done()
 
     describe 'when a test has been manually set to skip in a hook', () ->
+      clonedTransaction = null
 
-      beforeEach () ->
+      beforeEach (done) ->
         sinon.stub configuration.emitter, 'emit'
+
+        clonedTransaction = clone(transaction)
+
         runner = new Runner(configuration)
 
+        addHooks runner, [clonedTransaction], (err) ->
+          done err if err
+
+          runner.hooks.beforeHooks =
+            'Group Machine > Machine > Delete Message > Bogus example name' : [
+              (transaction) ->
+                transaction.skip = true
+            ]
+          done()
+
       afterEach () ->
         configuration.emitter.emit.restore()
 
       it 'should skip the test', (done) ->
-        transaction.skip = true
-        runner.executeTransaction transaction, () ->
+        runner.executeAllTransactions [clonedTransaction], runner.hooks, () ->
           assert.ok configuration.emitter.emit.calledWith 'test skip'
           done()
 
+      it 'should add skip message as a warning under `general` to the results on transaction', (done) ->
+        runner.executeAllTransactions [clonedTransaction], runner.hooks, () ->
+          messages = clonedTransaction['results']['general'].map (value, index) -> value['message']
+          assert.include messages.join(), 'skipped'
```

Comment:
please test for `'warning'` too, or add another `it` to test the severity `warning` (or add the warning filter into the `.map` function :100: